hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -80,17 +100,167 @@ public MapQueryEngineImpl(MapServiceContext mapServiceContext, QueryOptimizer op
         this.logger = nodeEngine.getLogger(getClass());
         this.queryResultSizeLimiter = new QueryResultSizeLimiter(mapServiceContext, logger);
         this.queryOptimizer = optimizer;
+        this.operationService = nodeEngine.getOperationService();
+        this.clusterService = nodeEngine.getClusterService();
+        this.localMapStatsProvider = mapServiceContext.getLocalMapStatsProvider();
+        this.parallelEvaluation = nodeEngine.getGroupProperties().getBoolean(QUERY_PREDICATE_PARALLEL_EVALUATION);
+        this.executor = nodeEngine.getExecutionService().getExecutor(QUERY_EXECUTOR);
+    }
+
+    public QueryResultSizeLimiter getQueryResultSizeLimiter() {
+        return queryResultSizeLimiter;
     }
 
     @Override
+    public QueryResult queryLocalPartitions(String mapName, Predicate predicate) throws ExecutionException, InterruptedException {
+        int initialPartitionStateVersion = partitionService.getPartitionStateVersion();
+        Collection<Integer> initialPartitions = mapServiceContext.getOwnedPartitions();
+        MapContainer mapContainer = mapServiceContext.getMapContainer(mapName);
+
+        // first we optimize the query
+        predicate = queryOptimizer.optimize(predicate, mapContainer.getIndexes());
+
+        // then we try to run using an index, but if that doesn't work, we'll try a full table scan
+        // This would be the point where a query-plan should be added. Is should determine if a full table scan
```

Comment:
It isn't really a todo. It is more a reminder that this is the place to be when this stuff gets added.